gcloud/datastore/query.py

```diff
@@ -463,6 +463,11 @@ def __iter__(self):
                 yield entity
             if not self._more_results:
                 break
+            num_results = len(self._page)
+            if self._limit is not None:
+                self._limit -= num_results
+            if self._offset is not None:
+                self._offset += num_results
```

Comment:
Ah yes, I understand your last comment now. The cursor will determine the progress made. The offset needs to be adjusted in the event that some of the offset hasn't been satisfied yet.